wgsl/index.bs

```diff
@@ -4557,11 +4555,282 @@ TODO: *Stub*: Expression evaluation
 
 ## Uniformity TODO ## {#uniformity}
 
-### Uniform control flow TODO ### {#uniform-control-flow}
+### Terminology and concepts TODO ### {#uniformity-concepts}
+
+The following definitions are merely informative, trying to give an intuition for what the analysis in the next subsection is computing.
+The analysis is what actually defines these concepts, and when a program is valid or break the uniformity rules.
+
+Control-flow is said to be uniform at a point in the program if all invocations execute in lockstep at that program point.
+An expression is said to be uniform if it is in uniform control-flow, and all invocations compute the same value for it.
+A local variable is said to be uniform, if all invocations have the same value in that variable at every program point where it is live.
```

Comment:
I think this is a little misleading, since it's not "same value" really, but rather the value is dependent on the same control flow in all invocations.

Reply:
I disagree with @jdashg's comment here. The way I use the term, a variable can be in uniform control flow and be non-uniform because it has different values across the scope of the uniformity. I would argue this definition because it is useful. If a control flow construct depends on a non-uniform variable under my definition, the control flow will be non-uniform. To express that in a definition where uniformity of a variable depends on the uniformity of its control flow, we would have to redefine a characteristic of variables to plug into that statement.

Furthermore, the uniformity of a uniform buffer concerns the persistence of its values across a broad scope. The same should apply to defining a variable as uniform.